service-worker/selective-caching/service-worker.js

```diff
@@ -29,20 +29,27 @@ var CURRENT_CACHES = {
   font: 'font-cache-v' + CACHE_VERSION
 };
 
+function hashValues(hash) {
+  return Object.keys(hash).map(function(key) { return hash[key]; });
+}
+
+function array2set(array) {
+  var set = {};
+  array.map(function(item) { set[item] = 1; });
+  return set;
+}
+
 self.addEventListener('activate', function(event) {
   // Delete all caches that aren't named in CURRENT_CACHES.
   // While there is only one cache in this example, the same logic will handle the case where
   // there are multiple versioned caches.
-  var expectedCacheNames = Object.keys(CURRENT_CACHES).map(function(key) {
-    return CURRENT_CACHES[key];
-  });
-
+  var expectedCacheNamesSet = array2set(hashValues(CURRENT_CACHES));
```

Comment:
I think we should just create a proper `new Set()` and use it here. `array2set` is uneeded